phantazm-commons/src/main/java/com/github/phantazmnetwork/commons/AdventureConfigProcessors.java

```diff
@@ -0,0 +1,125 @@
+package com.github.phantazmnetwork.commons;
+
+import com.github.steanky.ethylene.core.ConfigElement;
+import com.github.steanky.ethylene.core.ConfigPrimitive;
+import com.github.steanky.ethylene.core.collection.ConfigNode;
+import com.github.steanky.ethylene.core.collection.LinkedConfigNode;
+import com.github.steanky.ethylene.core.processor.ConfigProcessException;
+import com.github.steanky.ethylene.core.processor.ConfigProcessor;
+import net.kyori.adventure.key.InvalidKeyException;
+import net.kyori.adventure.key.Key;
+import net.kyori.adventure.sound.Sound;
+import net.kyori.adventure.text.Component;
+import net.kyori.adventure.text.minimessage.MiniMessage;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Objects;
+
+public final class AdventureConfigProcessors {
+    private AdventureConfigProcessors() {
+        throw new UnsupportedOperationException();
+    }
+
+    private static final ConfigProcessor<Key> key = new ConfigProcessor<>() {
+        @Override
+        public Key dataFromElement(@NotNull ConfigElement element) throws ConfigProcessException {
+            if(!element.isString()) {
+                throw new ConfigProcessException("Element must be a string");
+            }
+
+            try {
+                //noinspection PatternValidation
+                return Key.key(element.asString());
+            }
+            catch (InvalidKeyException keyException) {
+                throw new ConfigProcessException(keyException);
+            }
+        }
+
+        @Override
+        public @NotNull ConfigElement elementFromData(Key key) {
+            return new ConfigPrimitive(key.asString());
+        }
+    };
+
+    private static class ComponentConfigProcessor implements ConfigProcessor<Component> {
+
+        private final MiniMessage miniMessage;
+
+        private final boolean optional;
+
+        public ComponentConfigProcessor(@NotNull MiniMessage miniMessage, boolean optional) {
```

Comment:
passing in minimessage is #cursed :skull: when you're just assigning MiniMessage.miniMessage() anyway in a private static final decl

Reply:
custom MiniMessage instances can be created, idk if we'll ever need it, if not then ok to remove